### ISSUE TITLE
False positive "Method has no '__self__' member"

### ISSUE THREAD

**Issue by jwilk** [PyCQA/pylint#958]
### Current behavior

For this program
```python
print('moo'.__str__.__self__)
```
pylint reports:
```
E:  1, 0: Method '__str__' has no '__self__' member (no-member)
E:  1, 0: Method '__str__' has no '__self__' member (no-member)
```
But bound methods do have the `__self__` attribute.

### pylint --version output
```
No config file found, using default configuration
pylint 1.5.6, 
astroid 1.4.6
Python 3.5.2rc1 (default, Jun 13 2016, 09:33:26) 
[GCC 5.4.0 20160609]
```

**Comment by PCManticore** [comment]
This is fixed in pylint 2.0 / astroid 1.5.0 (the versions from the repositories). Please give them a try, we have a new capability in 2.0 through which we can understand special methods as these.

**Comment by zhihuiFan** [comment]
I run into a same issue, however  I can't find out pylint 2.0 in the branch of  https://github.com/PyCQA/pylint or astroid 1.5.0.   what I can find is astroid 2.0 and pylint 1.6.1.  Could you point me where I can find out the version you provided?  I'd like to have a try. 

> $ pylint --version
> pylint 1.6.1,
> astroid 1.4.7
> Python 3.5.2 (v3.5.2:4def2a2901a5, Jun 26 2016, 10:47:25)

**Comment by moylop260** [comment]
2.0.0 is master yet.
Try it
`pip install --upgrade git+https://github.com/PyCQA/astroid@master git+https://github.com/PyCQA/pylint@master`

**Comment by zhihuiFan** [comment]
thank @moylop260 ,  this resolved my issue!
